extensions/emmet/src/util.ts

```diff
@@ -380,30 +380,34 @@ export const allowedMimeTypesInScriptTag = ['text/html', 'text/plain', 'text/x-t
  * If position is inside a script tag of type template, then it will be parsed to find the inner HTML node as well
  */
 export function getHtmlFlatNode(documentText: string, root: FlatNode | undefined, offset: number, includeNodeBoundary: boolean): HtmlFlatNode | undefined {
-	const currentNode: HtmlFlatNode | undefined = <HtmlFlatNode | undefined>getFlatNode(root, offset, includeNodeBoundary);
+	let currentNode: HtmlFlatNode | undefined = <HtmlFlatNode | undefined>getFlatNode(root, offset, includeNodeBoundary);
 	if (!currentNode) { return; }
 
-	const isTemplateScript = currentNode.name === 'script' &&
-		(currentNode.attributes &&
-			currentNode.attributes.some(x => x.name.toString() === 'type'
+	// If the currentNode is a script one, first set up its subtree and then find HTML node.
+	if (currentNode.name === 'script' && currentNode.children.length === 0) {
```

Comment:
Why do we check if `children.length` is 0?

Reply:
removed it.

Reply:
@rzhao271 , I am actually adding this check back.

It handles the following case:
Let's say we run this command multiple times without making any edits to the content of the file.

`<script type="text/plain">
  <p></p>
</script>`

First time, in the HTML tree corresponding to the document-root, there will have no children for the script tag and we would create one children.

Second time, in the HTML tree corresponding  to the document-root, there will be one child for the script tag (The one we added in previous step). If we dont have this check, we would add another child which would make script node having 2 children (which is incorrect)

Third time, in the HTML tree corresponding  to the document-root, there will be two children for the script tag (The one  previous step). if we dont have this check, we would add another child which would make script node having 2 children (which again is incorrect as it should have 1 children only)


So if the file is not edited, looks like there is some cached HTML tree that is returned and running this command more that 1 times would result in wrong HTML tree structure (a one that does not correspond to the text in the file).

So adding a 0 check to ensure that we don't add children to the script-node that has already been set up. Alternative approach is to clear the children array of script-node before setting up its subtree but that would again be not as performant as this check.